elevation_platform_node:
  ros__parameters:
    sim: False

/elevation_joint_motor_1/joint_motor_driver_node:
  ros__parameters:
    can_id: 0x60
    gear_ratio: 101.0

    # CAUTION: position_offset is defind as home position!
    position_offset: 0xFFFE3900 # Recorded in 20250703

    position_p: 800
    position_i: 0 # not used
    position_d: 500 
    velocity_p: 1100
    velocity_i: 1000
    velocity_d: 0  # not used

    max_forward_current: 10000 # mA
    min_backward_current: -10000 # mA
    max_forward_acceleration: 5.0 # FIXME
    max_backward_acceleration: 5.0 # FIXME
    max_forward_velocity: 5.0 # degree/s
    min_backward_velocity: -5.0 # degree/s

    auto_halt_timeout: 3 # seconds

/elevation_joint_motor_2/joint_motor_driver_node:
  ros__parameters:
    can_id: 0x61
    gear_ratio: 101.0

    # CAUTION: position_offset is defind as home position!
    position_offset: 0xF8CC # Recorded in 20250703

    position_p: 800
    position_i: 0 # not used
    position_d: 500 
    velocity_p: 1100
    velocity_i: 1000
    velocity_d: 0  # not used

    max_forward_current: 10000 # mA
    min_backward_current: -10000 # mA
    max_forward_acceleration: 5.0 # FIXME
    max_backward_acceleration: 5.0 # FIXME
    max_forward_velocity: 5.0 # degree/s
    min_backward_velocity: -5.0 # degree/s

    auto_halt_timeout: 3 # seconds

/elevation_joint_motor_3/joint_motor_driver_node:
  ros__parameters:
    can_id: 0x62
    gear_ratio: 101.0

    # CAUTION: position_offset is defind as home position!
    position_offset: 0x308C # Recorded in 20250703

    position_p: 800
    position_i: 0 # not used
    position_d: 500 
    velocity_p: 1100
    velocity_i: 1000
    velocity_d: 0  # not used

    max_forward_current: 10000 # mA 
    min_backward_current: -10000 # mA 
    max_forward_acceleration: 5.0 # FIXME
    max_backward_acceleration: 5.0 # FIXME
    max_forward_velocity: 5.0 # degree/s
    min_backward_velocity: -5.0 # degree/s

    auto_halt_timeout: 3 # seconds

/elevation_joint_motor_4/joint_motor_driver_node:
  ros__parameters:
    can_id: 0x63
    gear_ratio: 101.0

    # CAUTION: position_offset is defind as home position!
    position_offset: 0x19816 # Recorded in 20250703

    position_p: 800
    position_i: 0 # not used
    position_d: 500 
    velocity_p: 1100
    velocity_i: 1000
    velocity_d: 0  # not used
    
    max_forward_current: 10000 # mA 
    min_backward_current: -10000 # mA 
    max_forward_acceleration: 5.0 # FIXME
    max_backward_acceleration: 5.0 # FIXME
    max_forward_velocity: 45.0 # degree/s
    min_backward_velocity: -45.0 # degree/s

    auto_halt_timeout: 3 # seconds